test/DurableTask.SqlServer.Tests/Integration/DatabaseManagement.cs

```diff
@@ -163,37 +186,63 @@ public void SchemaCreationIsSerializedAndIdempotent()
             ValidateDatabaseSchema(testDb);
 
             // Operations are expected to be serialized, making the log output deterministic.
-            LogAssert.Sequence(
-                this.logProvider,
-                // 1st
-                LogAssert.AcquiredAppLock(statusCode: 0),
-                LogAssert.SprocCompleted("dt._GetVersions"),
-                LogAssert.ExecutedSqlScript("schema-0.2.0.sql"),
-                LogAssert.ExecutedSqlScript("logic.sql"),
-                LogAssert.ExecutedSqlScript("permissions.sql"),
-                LogAssert.SprocCompleted("dt._UpdateVersion"),
-                // 2nd
-                LogAssert.AcquiredAppLock(statusCode: 1),
-                LogAssert.SprocCompleted("dt._GetVersions"),
-                // 3rd
-                LogAssert.AcquiredAppLock(statusCode: 1),
-                LogAssert.SprocCompleted("dt._GetVersions"),
-                // 4th
-                LogAssert.AcquiredAppLock(statusCode: 1),
-                LogAssert.SprocCompleted("dt._GetVersions"));
+            LogAssert
+                .For(this.logProvider)
+                .Expect(
+                    // At least 1 worker will check the database first
+                    LogAssert.CheckedDatabase())
+                .Contains(
+                    // The other 3 workers will check in some non-deterministic order
+                    LogAssert.CheckedDatabase(),
+                    LogAssert.CheckedDatabase(),
+                    LogAssert.CheckedDatabase())
+                .ContainsIf(
+                    // One worker may obtain the lock after another worker created the database.
+                    isDatabaseMissing,
+                    LogAssert.CommandCompleted($"CREATE DATABASE [{testDb.Name}]"),
+                    LogAssert.CreatedDatabase(testDb.Name))
+                .OptionallyContainsIf(
+                    // Anywhere from 0 to 3 of the workers may end up attempting to create the database.
+                    isDatabaseMissing,
+                    LogAssert.CommandCompleted($"CREATE DATABASE [{testDb.Name}]"),
+                    LogAssert.CommandCompleted($"CREATE DATABASE [{testDb.Name}]"),
+                    LogAssert.CommandCompleted($"CREATE DATABASE [{testDb.Name}]"))
+                .Expect(
+                    // 1st
+                    LogAssert.AcquiredAppLock(statusCode: 0),
+                    LogAssert.SprocCompleted("dt._GetVersions"),
+                    LogAssert.ExecutedSqlScript("schema-0.2.0.sql"),
+                    LogAssert.ExecutedSqlScript("logic.sql"),
+                    LogAssert.ExecutedSqlScript("permissions.sql"),
+                    LogAssert.SprocCompleted("dt._UpdateVersion"),
+                    // 2nd
+                    LogAssert.AcquiredAppLock(),
```

Comment:
For the other workers, I don't think we can guarantee their response codes other than success; there may or may not be contention for the lock.